src/containers/chat/chat-text-area/ChatTextArea.tsx

```diff
@@ -13,43 +25,105 @@ import { spliceSx } from '~/utils/helper-functions'
 
 interface ChatTextAreaProps extends Omit<TextFieldProps, 'onChange' | 'sx'> {
   value: string
-  onChange: (e: ChangeEvent<HTMLInputElement>) => void
+  setValue: Dispatch<SetStateAction<string>>
   onClick: () => void
   sx?: {
     textAreaWrapper?: SxProps
     container?: SxProps
     icon?: SxProps
   }
+  emojiPickerProps?: { perLine: number }
 }
 
 const ChatTextArea: FC<ChatTextAreaProps> = ({
   value,
-  onChange,
+  setValue,
   onClick,
   maxRows = 6,
   minRows = 1,
   sx = {},
+  emojiPickerProps,
   ...props
 }) => {
+  const { isMobile } = useBreakpoints()
+  const [isEmojiPickerOpen, setIsEmojiPickerOpen] = useState<boolean>(false)
+  const inputRef = useRef<HTMLInputElement | null>(null)
+
+  const onChange = (e: ChangeEvent<HTMLInputElement>) =>
+    setValue(e.target.value)
+
+  const onEmojiSelect = ({ native: emoji }: { native: string }) => {
```

Comment:
refactor and move it to `~types`

Reply:
I thought about it, but I don't really see the point, I guess it will be used only here